napari/plugins/npe2api.py

```diff
@@ -46,9 +55,21 @@ def _user_agent() -> str:
 class SummaryDict(TypedDict):
     """Objects returned at https://npe2api.vercel.app/api/extended_summary ."""
 
-    name: PyPIname
+    name: NotRequired[PyPIname]
+    version: str
+    display_name: NotRequired[str]
+    summary: str
+    author: str
+    license: str
+    home_page: str
+    pypi_versions: NotRequired[List[str]]
+    conda_versions: NotRequired[List[str]]
+
+
+class _SummaryDict(TypedDict):
+    """Objects returned at https://npe2api.vercel.app/api/extended_summary ."""
```

Comment:
Because we pop name from the original dict and pass it as a separate argument that is to hard for mypy to discover. But I may try to change the logic to be simpler instead of only adjusting type annotation. 

What did you think?

Reply:
`_SummaryDict` was removed.